### PR TITLE
Imx9 bug fixes 2511

### PR DESCRIPTION
**Pull request by joukkone**

### CODE REVIEW

File: arch/arm64/src/imx9/imx9_flexspi_nor.c [hunk 1]
```diff
@@ -790,12 +790,15 @@ static ssize_t imx9_flexspi_nor_read(struct mtd_dev_s *dev,
     }
 
   src = priv->ahb_base + offset;
-  DEBUGASSERT(((uintptr_t)src & ALIGN_MASK) == 0);
 
   up_invalidate_dcache((uintptr_t)buffer,
                        (uintptr_t)buffer + ALIGN_UP(nbytes));
+  int n = nbytes;
```
Comment (jlaitine) [inline arch/arm64/src/imx9/imx9_flexspi_nor.c:797]: This is fine, but what is the cache invalidate above about???

File: arch/arm64/src/imx9/imx9_flexspi_nor.c [hunk 2]
```diff
@@ -790,12 +790,15 @@ static ssize_t imx9_flexspi_nor_read(struct mtd_dev_s *dev,
...
 
-  memcpy(buffer, src, nbytes);
+  while (n-- > 0)
+    {
+      *buffer++ = *src++;
+    }
```
Comment (jlaitine) [inline arch/arm64/src/imx9/imx9_flexspi_nor.c:801]: Again, the cache clean below? Should we remove those at the same if they are useless?

**Review by jlaitine — APPROVED** [review]
Lgtm, I added comments about some other weirdness

File: arch/arm64/src/imx9/imx9_flexspi_nor.c (hunk 2 above)
Comment (joukkone) [reply arch/arm64/src/imx9/imx9_flexspi_nor.c:799]: done

File: arch/arm64/src/imx9/imx9_flexspi_nor.c (hunk 1 above)
Comment (joukkone) [reply arch/arm64/src/imx9/imx9_flexspi_nor.c:795]: done, removed

**Pull request merged**